docs/alerting/api.md

```diff
@@ -858,3 +858,281 @@ _opendistro/_alerting/destinations/<destination-id>
+#### Request
+```
+PUT _opendistro/_alerting/destinations/email_accounts/<email_account_id>
+``` 
+
+## Get email account
+
+#### Request
+```
+GET _opendistro/_alerting/destinations/email_accounts/<email_account_id>
+```
+
+## Delete email account
+
+#### Request
+```
+DELETE _opendistro/_alerting/destinations/email_accounts/<email_account_id>
+```
+#### Sample response
+
+```json
+{
+  "_index" : ".opendistro-alerting-config",
+  "_type" : "_doc",
+  "_id" : "<email_account_id>",
+  "_version" : 1,
+  "result" : "not_found",
+  "forced_refresh" : true,
+  "_shards" : {
+    "total" : 2,
+    "successful" : 2,
+    "failed" : 0
+  },
+  "_seq_no" : 12,
+  "_primary_term" : 2
+}
+```
+
+## Search email account
+
+#### Request
+```json
+POST _opendistro/_alerting/destinations/email_accounts/_search
+
+{
+  "from": 0,
+  "size": 20,
+  "sort": { "email_account.name.keyword": "desc" },
+  "query": {
+    "bool": {
+      "must": {
+        "match_all": {}
+      }
+    }
+  }
+}
+```
+
+#### Sample response
+
+```json
+{
+  "took" : 8,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 2,
+      "relation" : "eq"
+    },
+    "max_score" : null,
+    "hits" : [
+      {
+        "_index" : ".opendistro-alerting-config",
+        "_type" : "_doc",
+        "_id" : "RAau5XQBBF7ehDf33O3G",
+        "_seq_no" : 8,
+        "_primary_term" : 2,
+        "_score" : null,
+        "_source" : {
+          "schema_version" : 2,
+          "name" : "example_account",
+          "email" : "example@email.com",
+          "host" : "smtp.email.com",
+          "port" : 465,
+          "method" : "ssl"
+        },
+        "sort" : [
+          "example_account"
+        ]
+      }
+```
+
+---
+ 
+## Create email group
+
+#### Request
+
+```json
+POST _opendistro/_alerting/destinations/email_groups
+{
+  "name": "example_email_group",
+  "emails": [{
+    "email": "example@email.com"
+  }]	
+}
+```
+
+#### Sample response
+
+```json
+{
+  "_id" : "EQbQ5XQBBF7ehDf3CO7_",
+  "_version" : 1,
+  "_seq_no" : 9,
+  "_primary_term" : 2,
+  "email_group" : {
+    "schema_version" : 2,
+    "name" : "example_email_group",
+    "emails" : [
+      {
+        "email" : "example@email.com"
+      }
+    ]
+  }
+}
+```
+
+## Update email group
+
+#### Request
+
+```json
+PUT _opendistro/_alerting/destinations/email_groups/<email_group_id>
+{
+  "name": "example_email_group",
+  "emails": [{
+    "email": "example@email.com"
+  }]	
+}
+```
+ 
+## Get email group
+
+#### Request
+```
+GET _opendistro/_alerting/destinations/email_groups/<email_group_id>
+```
+ 
+## Delete email group
+
+#### Request
+```
+DELETE _opendistro/_alerting/destinations/email_groups/<email_group_id>
+```
+#### Sample response
+
+```json
+{
+  "_index" : ".opendistro-alerting-config",
+  "_type" : "_doc",
+  "_id" : "<email_group_id>",
+  "_version" : 1,
+  "result" : "not_found",
```

Comment:
Same for this one as well, if we're posting the expected responses of successful scenarios, this should be:

```
"result" : "deleted"
```

Reply:
done

Reply:
done